### DISCUSSION TITLE
is it intentional that we never want to disable ColorPicker or make it readonly ?

### DISCUSSION THREAD

**Discussion by YashodipD** [iTwin/iTwinUI#2203] [Q&A]
Currently I have added ColorPicker to form. It works great however I have a situation where I want to disable ColorPicker or make it readable only in certain situation.

**Comment by mayank99** [comment]
Generally speaking, disabled controls are an anti-pattern, and the concept of a "disabled ColorPicker" doesn't make sense, because it currently only provides the UI *inside* the popover. But if you really want to disable the ColorPicker, you should disable its triggering button, so that the popover doesn't even show up.

```jsx
<Popover content={<ColorPicker />}>
  <Button disabled>Pick color</Button>
</Popover>
```
